homeassistant/components/torque/sensor.py

```diff
@@ -42,21 +44,64 @@ def convert_pid(value):
     return int(value, 16)
 
 
-def setup_platform(
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor from a config entry created in the integrations UI."""
+    vehicle = config_entry.data.get(CONF_NAME)
+    email = config_entry.data.get(CONF_EMAIL)
+    sensors: dict[int, TorqueSensor] = {}
+
+    hass.http.register_view(
+        TorqueReceiveDataView(email, vehicle, sensors, async_add_entities)
+    )
```

Comment:
What's this?

Reply:
This is from the original implementation. It's just the handler for the `/api/torque` API route.

Reply:
But I am just curious in general, what does it do? Do you have a screenshot of what it does?

Reply:
I don't really have a screenshot, but basically it just receives data (similar to a webhook), except Torque sends data as query parameters. The query parameters are like `userFullName{pid}`, `userUnit{pid}`, and `k{pid}` (where `pid` is a hex-encoded integer, such as `0x0c` for Engine RPM).

Basically something like this:
```http
GET http://localhost:8123/api/torque?eml=me@test.local&userFullName0x0c=Engine%20RPM&userUnit0x0c=RPM&k0x0c=1
```

This request handler just formats those, and converts them to HA entities.

<img width="543" alt="Screenshot 2024-01-04 at 22 27 23" src="https://github.com/home-assistant/core/assets/1899334/1f95cf17-82d7-4f56-9ccd-ab0154a8b2a8">